larpandoracontent/LArReclustering/CheatedThreeDClusteringTool.cc

```diff
@@ -0,0 +1,102 @@
+/**
+ *  @file   larpandoracontent/LArReclustering/CheatedThreeDClusteringTool.cc
+ *
+ *  @brief  Implementation file for the cheated threeD reclustering algorithm.
+ *
+ *  $Log: $
+ */
+
+#include "Pandora/AlgorithmHeaders.h"
+#include "larpandoracontent/LArHelpers/LArMCParticleHelper.h"
+
+#include "larpandoracontent/LArReclustering/CheatedThreeDClusteringTool.h"
+
+
+using namespace pandora;
+
+namespace lar_content
+{
+
+CheatedThreeDClusteringTool::CheatedThreeDClusteringTool()
+{
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+bool CheatedThreeDClusteringTool::Run(const Algorithm *const pAlgorithm, std::vector<pandora::CaloHitList*> &newCaloHitListsVector)
+{
+    if (newCaloHitListsVector.size() != 1)
+        return false;
+
+    CaloHitList *initialCaloHitList = newCaloHitListsVector.at(0);
+    newCaloHitListsVector.clear();
+
+    std::map<int, CaloHitList*> McIdCaloHitListMap;
```

Comment:
I had this as a pointer because it then get pushed into newCaloHitListsVector, which is: std::vector<pandora::CaloHitList*> &newCaloHitListsVector
I could otherwise remove the pointer/use the address. Is that preferred?

Reply:
If there's a good reason for it being a pointer then it should definitely stay as a pointer.  If its purely a means of avoiding copies then a reference (and std::reference_wrapper when storing in a container) is probably safer)

Reply:
Yes. I suspect this choice followed from some of the use cases in Pandora where pointers to lists get passed around for the purposes of Pandora's list management, but I don't think any of that is required here, because this part of the code isn't interacting with the list managers - I also suggested not reusing the input list in this way.

Reply:
This has changed (as per comment above: I had a go at restructuring the code. This compiles but needs to be tested, as there are quite a few changes.)